src/components/Main/Schedule/Home.tsx

```diff
@@ -1,51 +1,107 @@
 import React, { useState, useEffect } from 'react';
 import { useDispatch } from 'react-redux';
-import { useQuery } from 'react-query';
+import { useQuery, useMutation, useQueryCache } from 'react-query';
 
 import { Scheduler } from './Scheduler';
 import { ScheduleTabs } from './ScheduleTabs';
+import { NewScheduleForm } from './NewScheduleForm';
+import { NewServiceForm } from './NewServiceForm';
 
 import { logout } from '../../../store/actions';
 import { useSelector } from '../../../shared/utilities';
 import { getScheduleData } from '../../../query/schedules';
+import { addSchedule, addService } from '../../../store/apis/schedules';
 
+import { Dialog } from '@material-ui/core';
 import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
 import { buttonTheme } from '../../../shared/styles/theme.js';
 import { useSpinner } from '../../../shared/styles/loading-spinner';
 
+import AddIcon from '@material-ui/icons/Add';
+
 export const Home = () => {
   const classes = useStyles();
-  useSpinner();
+  // useSpinner();
 
   const dispatch = useDispatch();
 
   // React-query
+  const cache = useQueryCache();
   const { churchId, name: churchName } = useSelector((state) => state.profile);
   const { isLoading, error, data = [] } = useQuery(
     ['schedulesData', churchId],
     getScheduleData,
     {
       enabled: churchId,
       refetchOnWindowFocus: false,
-      staleTime: 100000000000000,
+      staleTime: 100000000000000, //1157407.4 days fyi lol
     },
   );
+  const [mutateAddSchedule] = useMutation(addSchedule, {
+    onSuccess: () => cache.invalidateQueries('schedulesData'), //causes the schedulesData query to call and update on success
+  });
+  const [mutateAddService] = useMutation(addService, {
+    onSuccess: () => cache.invalidateQueries('schedulesData'),
+  });
 
-  if (status === 'loading') return <div>loading...</div>; // loading state
+  // if (status === 'loading') return <div>loading...</div>; // replaced by useSpinner() ??
```

Comment:
yeah prolly can just delete this line. i remember trying to get this loading div to show up, but it doesn't show up on my machine.